### ISSUE TITLE
ModelAdmin InspectView raises SourceImageIOError

### ISSUE THREAD

**Issue by robslotboom** [wagtail/wagtail#3263]
### Issue Summary

When enabling InspectView for a ModelAdmin the InspectView raises the above error when the object inspected contains an image.

I guess this is because Dynamic image serve view is required.

### Steps to Reproduce

Create a custom model containing a fk to wagtailimage.Image:

class MyModel(models.Model):
    image = models.ForeignKey('wagtailimages.Image')

Create a custom ModelAdmin

class MyModelAdmin(ModelAdmin):
	model = MyModel
	inspect_view_enabled = True

Register it:
modeladmin_register(MyModelAdmin)

Now create an instance using Wagtail.
After submitting klik on the inspect button.

### Technical details

* Python Python 3.5.0
* Django 1.10.1
* Wagtail 1.8

**Comment by ababic** [comment]
Hi @robslotboom,

Are you able to confirm whether you are still having this issue? And is it only raised when trying to load `InspectView` for a model instance, or are you having issues with images in general?

The only place within `InspectView` that could be causing this issue is the [`get_image_field_display ()`](https://github.com/wagtail/wagtail/blob/2c9805dc3fa37e9bf27fc7593de4465673faffca/wagtail/contrib/modeladmin/views.py#L888) method.  @gasman, can you see anything obviously wrong with the code for that method?

**Comment by robslotboom** [comment]
Hi ababic,

Yes, only on InspectView for a model instance.

**Comment by ababic** [comment]
I'm admittedly not that great with environment management/setup, so haven't been able to configure something using the same Python version (i'm using `2.7.10` here), but with a fresh project (following Wagtail's quickstart instructions), and simple Model/ModelAdmin set up as explained, I am not seeing the same issue.

![Uploading Screenshot 2017-01-21 at 13.12.43.png…]()

 Maybe there's something in the Python version, but I'd be surprised if there were.

Have you tried with different images / image types, does the issue persist if you reupload the original image?

**Comment by robslotboom** [comment]
Hi ababic,

I enabled the InspectView again and now it is working ok. I guess because I added ServeView to the urls in the meantime.

Commenting it out isn’t an option because some methods I wrote depend on it.

**Comment by ababic** [comment]
Okay, glad the issue has resolved itself 👍

**Comment by robslotboom** [comment]
Maybe the lack of the ServeView in urls is causing the error. Unfortunately I can’t test it.

**Comment by ababic** [comment]
Hi @robslotboom. Use of `wagtailimages.views.serve.ServeView` in a project shouldn't effect `InspectView`. The method that renders images for appropriate field values in `InspectView` calls the `get_rendition()` method on the `Image` model directly, so ServeView isn't a part of the process. 

Having looked to see how the `{% image %}` template tag works, I've just discovered the `wagtailimages.shortcuts.get_rendition_or_not_found()` method, which looks like what `InspectView` should be using, but isn't (I must've missed that when I wrote InspectView initially). It looks like that method handles missing source images in a more graceful way (which is what I think is the cause of your original issue).

I'll work on a PR soon.
